### PR TITLE
chore: deprecate tox in favor of act

### PR DESCRIPTION
**Pull request by mistercrunch**
Set up `act` as a mean to run GitHub Actions locally as an alternative to `tox`.

- deprecating `tox`
- documenting how to use `act`

### CODE REVIEW

File: requirements/development.txt [hunk 1]
```diff
@@ -6,16 +6,16 @@
 #    pip-compile-multi
 #
 -r base.txt
--e file:.
+-e file:///Users/max/code/superset
```
Comment (john-bodley) [inline requirements/development.txt:9]:
```suggestion
-e file:.
```

Comment (mistercrunch) [reply requirements/development.txt:9]: oh yeah that's a `pip-compile-multi` bug AFACT, it shows randomly every 2-4 runs for me... Not sure if it's been reported.

**Comment by villebro** [comment]
Interesting, I tried this a few years ago iIRC, and it was horribly buggy. Will take it for a new spin..

**Comment by betodealmeida** [comment]
> Interesting, I tried this a few years ago iIRC, and it was horribly buggy. Will take it for a new spin..

Same, I tried using it in 2020 and it was unusable! It'll be good to have a way of running the CI locally, assuming the runs are representative.

**Review by betodealmeida — APPROVED** [review]

**Comment by rusackas** [comment]
Same, I could never get `act` to _actually_ work. I'll pull/test this, and if I can have luck, I'll merge it AND buy you a beer.

**Comment by mistercrunch** [comment]
Let me try and make sure most workflows that someone might want to run locally actually runs through.

**Comment by mistercrunch** [comment]
@rusackas seems like --secret and --container-architecture fixed a lot of things for me, not sure how much you pushed but I have to admit I haven't tried anywhere close to all the workflows yet. Actually many of the workflows don't make sense to run locally...
```bash
act --job {workflow_name} --secret GITHUB_TOKEN=$GITHUB_TOKEN --event pull_request --container-architecture linux/amd64
```

**Comment by rusackas** [comment]
As we discussed on Zoom, I'm still having a hard time getting any of this to work... 
* it **is** having trouble if you've done `npm install` in `superset-frontend`, `cypress-base`, `docs`, `superset-websocket` etc, as it finds various `travis.yml` files and gets confused.
* even if you `git clean -fdx` to get back to a cleared repo, I still get issues. * it _might_ be having some trouble if your git origin is https (tbd)

Hopefully there's a way to at least get around the first issue by pointing it to the workflow files(s) it should pay attention to.  I'll do a little reading and see if I get lucky.

**Comment by mistercrunch** [comment]
Realizing there's significantly much more work to do here. Last time I pushed this forward I found some clear incompatibility between the base Ubuntu images used by github and the ones used by `act`, something around python's SSL not being configured right in the catthehacker/act-ubuntu image

**Comment by mistercrunch** [comment]
Thinking about the success criteria for this PR, seems 

1. we should be able to run all "locally-relevant" GHA workflow. This means stuff like linting, running tests backend/frontend, pre-commit all, ... We don't need for workflows like the ephemeral envs ones and such to work locally.

2. should be relatively easy to set up, easier than `tox` at least

**Comment by rusackas** [comment]
Should we move this back to `draft` mode until some of the dust settles?

**Comment by mistercrunch** [comment]
Yup, also would welcome some help here as I got tangled up in other things and it looks like this is getting pushed to the bottom of my todo list now. Things seem super promising, but there's quite a few little things to connect. I discovered there's support for `.env` files an `.secrets` files which both can reference env vars and set defaults. Seemed powerful.

The last blocker I had was around the `pip install` not working (because python ssl lib not properly set up, and while trying to reach out to http**S**://pypy.org ) on the base docker images that `act` is using. Seems pretty doable to swap to any other image, but I didn't get too far.

File: docs/docs/contributing/development.mdx [hunk 2]
```diff
@@ -502,39 +502,117 @@ If using the eslint extension with vscode, put the following in your workspace `
...
 ```
 
-## Testing
 
-### Python Testing
+## GitHub Actions and `act`
+
+:::tip
+`act` compatibility of Superset's GHAs is not fully tested. Running `act` locally may or may not
+work for different actions, and may require fine tunning and local secret-handling.
```
Comment (rusackas) [inline docs/docs/contributing/development.mdx:510]:
```suggestion
work for different actions, and may require fine tuning and local secret-handling.
```

File: docs/docs/contributing/development.mdx [hunk 3]
```diff
@@ -502,39 +502,117 @@ If using the eslint extension with vscode, put the following in your workspace `
...
+database engines (PostgreSQL, MySQL, SQLite) and different versions of Python.
+This enables us to ensure compatibility and stability across various environments.
+:::
 
-All python tests are carried out in [tox](https://tox.readthedocs.io/en/latest/index.html)
-a standardized testing framework.
-All python tests can be run with any of the tox [environments](https://tox.readthedocs.io/en/latest/example/basic.html#a-simple-tox-ini-default-environments), via,
+### Lower-Level Control
+
+Note that for all GHAs, it's also possible to go lower level and the content of any workflow
```
Comment (rusackas) [inline docs/docs/contributing/development.mdx:544]: I'm not sure I understand this sentence.

**Comment by mistercrunch** [comment]
Not everything works here, but no one uses `tox` anymore AFAIK, and I think the setup is broken. I advocate for merging this and polishing/improving `act` and `docker-compose` support over time

**Review by rusackas — APPROVED** [review]

**Comment by rusackas** [comment]
I, for one, will not miss tox. This'll be a much better DevEx.

**Comment by jpadams** [comment]
@mistercrunch and @rusackas I've had a todo to start work on a Dagger PR for Superset for the last six months, but haven't started yet. Bravo on making things better!

**Pull request merged**